src/blockdata/transaction.rs

```diff
@@ -36,28 +36,59 @@ use network::serialize::{serialize, BitcoinHash, SimpleEncoder, SimpleDecoder};
 use network::encodable::{ConsensusEncodable, ConsensusDecodable, VarInt};
 
 /// A reference to a transaction output
-#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Debug, Hash)]
-pub struct TxOutRef {
+#[derive(Copy, Clone, Debug, Eq, Hash, PartialEq, PartialOrd, Ord)]
+pub struct OutPoint {
     /// The referenced transaction's txid
     pub txid: Sha256dHash,
     /// The index of the referenced output in its transaction's vout
-    pub index: usize
+    pub vout: u32,
 }
-serde_struct_impl!(TxOutRef, txid, index);
+serde_struct_impl!(OutPoint, txid, vout);
 
-impl fmt::Display for TxOutRef {
+impl OutPoint {
+    /// Creates a "null" `OutPoint`.
+    ///
+    /// This value is used for coinbase transactions because they don't have
+    /// any previous outputs.
+    #[inline]
+    pub fn null() -> OutPoint {
+        OutPoint {
+            txid: Default::default(),
+            vout: u32::max_value(),
+        }
+    }
```

Comment:
Can you make this function be the `Default::default` implementation of the `Outpoint` type? You can also have the explicit method.